package.json

```diff
@@ -51,7 +52,7 @@
     "eslint": "^8.25.0",
     "ethereum-waffle": "^3.4.0",
     "ethers": "^5.4.1",
-    "hardhat": "^2.10.2",
+    "hardhat": "^2.12.0",
```

Comment:
Wanted to call out the Hardhat version upgrade - I had to switch versions to fix a typeerror that Smock was giving me (`Cannot read properties of undefined (reading 'events')`) but please lmk if we wanted to stay consistent with the earlier version 🙏🏻

Reply:
This SGTM 🙏